pulp_container/app/registry_api.py

```diff
@@ -349,6 +325,32 @@ def get_dr_push(self, request, path, create=False):
                 raise RepositoryNotFound(name=path)
         return distribution, repository
 
+    @transaction.atomic
```

Comment:
there is a different wrapping into a transaction whole method vs just the try block

Reply:
I removed this change. The logic should be similar to the previous one, where we catch exceptions outside the `transaction.atomic()`'s context manager.